workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerCoordinatorWorkerConnector.cs

```diff
@@ -48,20 +47,29 @@ protected override async void HandleWorkerConnectionEstablished()
 
         if (SimulatedPlayerWorkerConnector != null)
         {
-            while (simulatedPlayerConnectors.Count < DefaultSimulatedPlayerCount)
+            while (Application.isPlaying && simulatedPlayerConnectors.Count < DefaultSimulatedPlayerCount)
             {
-                await Task.Delay(TimeSpan.FromSeconds(
-                    Random.Range(DefaultSimulatedPlayerCreationInterval, 1.25f * DefaultSimulatedPlayerCreationInterval)));
                 var simulatedPlayer = Instantiate(SimulatedPlayerWorkerConnector, transform.position, transform.rotation);
                 await simulatedPlayer.GetComponent<SimulatedPlayerWorkerConnector>()
                     .ConnectSimulatedPlayer(Worker.LogDispatcher, SimulatedPlayerDevAuthTokenId, SimulatedPlayerTargetDeployment);
                 simulatedPlayerConnectors.Add(simulatedPlayer);
+                await Task.Delay(TimeSpan.FromSeconds(
+                    Random.Range(DefaultSimulatedPlayerCreationInterval, 1.25f * DefaultSimulatedPlayerCreationInterval)));
             }
 
-            StartCoroutine(MonitorSimulatedPlayers());
+            if (Application.isPlaying)
+            {
+                StartCoroutine(MonitorSimulatedPlayers());
+            }
         }
     }
 
+    public override void Dispose()
```

Comment:
decided to move it to `Dispose` instead of `OnDestroy`to keep it consistent with the other connectoes where we overrode `Dispose` instead of `OnDestroy` - also means that this branch is not dependent on the other branch in the gdk repo anymore